matplotcheck/raster.py

```diff
@@ -140,27 +184,12 @@ def assert_legend_accuracy_classified_image(
         # Check that each legend entry label is in one of all_label_options
         assert len([val for val in legend_dict.values() if val]) == len(
             all_label_options
-        ), "Incorrect legend labels"
+        ), (
+            "Number of label options provided doesn't match the number of"
+            " labels found in the image."
+        )
 
-        # Create two copies of image array, one filled with the plot data class
-        # labels (im_data_labels) and the other with the expected labels
-        # (im_expected_labels)
-        im_class_dict = {}
-        for val in np.unique(im_data):
```

Comment:
this is the part i'm wondering about - does this map cmap values to colors and labels?